shadowsocks/tcp.go

```diff
@@ -218,14 +207,12 @@ func (s *tcpService) Start() {
 					logger.Debugf("TCP Error: %v: %v", connError.Message, connError.Cause)
 					status = connError.Status
 				}
-				s.m.AddClosedTCPConnection(clientLocation, keyID, status, proxyMetrics, timeToCipher, connDuration)
+				s.m.AddClosedTCPConnection(clientLocation, keyID, status, proxyMetrics, connDuration)
 				clientConn.Close() // Closing after the metrics are added aids integration testing.
 				logger.Debugf("Done with status %v, duration %v", status, connDuration)
 			}()
 
-			findStartTime := time.Now()
-			keyID, clientConn, salt, err := findAccessKey(clientConn, s.ciphers)
-			timeToCipher = time.Now().Sub(findStartTime)
+			keyID, clientConn, salt, err := findAccessKey(clientConn, s.ciphers, s.m)
```

Comment:
Let's record the time to cipher metric here.

Reply:
I reverted this to the code from master.